### PR TITLE
Tpcds

### PR DESCRIPTION
**Pull request by KishorBachhav**
## Changes proposed in this pull request

TPCDS implementation  …
- Two separate jobs
  1> For creating Table
  2> For executing query
-Made certain properties configurable like number of buckets, warm up and average run iterations

## Patch testing

Ran:
 TPCDS snappy jobs

## ReleaseNotes.txt changes

(Does this change require an entry in ReleaseNotes.txt? If yes, has it been added to it?)

## Other PRs 

(Does this change require changes in other projects- store, spark, spark-jobserver, aqp? Add the links of PR of the other subprojects that are related to this change)

### CODE REVIEW

File: cluster/src/test/scala/io/snappydata/benchmark/snappy/TPCDSSuite.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ import org.apache.spark.{SparkConf, SparkContext}
+ import org.apache.spark.sql.{Row, SnappyContext, SnappySession, SparkSession}
+ import org.apache.spark.sql.execution.benchmark.{TPCDSQueryBenchmark, TPCDSQuerySnappyBenchmark}
+ import org.scalatest.{BeforeAndAfterAll, FunSuite}
+
+ import scala.collection.mutable.ArrayBuffer
+
+
+/**
+  * Created by sbhokare on 7/7/17.
```
Comment (sonalsagarwal) [inline cluster/src/test/scala/io/snappydata/benchmark/snappy/TPCDSSuite.scala]: Author section not required to be checked in

File: cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/QueryExecutionJob.scala [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import com.typesafe.config.Config
+import io.snappydata.benchmark.TPCH_Queries
+import io.snappydata.benchmark.snappy.tpch.QueryExecutor
+
+import org.apache.spark.sql.catalyst.util.fileToString
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{Row, SnappyJobInvalid, SnappyJobValid, SnappyJobValidation, SnappySQLJob, SnappySession}
+
+/**
+  * Created by kishor on 10/1/18.
```
Comment (sonalsagarwal) [inline cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/QueryExecutionJob.scala]: Author section needs not to be checked-in. Can you remove from all files, wherever it exists.

File: tests/benchmark/snappy/tpcds/spark/snappydata.log [hunk 3]
```diff
@@ -0,0 +1,720 @@
+18/01/12 13:00:34.439 IST main<tid=0x1> INFO SparkContext: Running Spark version 2.1.1.1
```
Comment (sonalsagarwal) [inline tests/benchmark/snappy/tpcds/spark/snappydata.log]: log file should not be checked-in.

File: tests/benchmark/snappy/tpcds/PerfRun.conf [hunk 4]
```diff
@@ -0,0 +1,60 @@
+#Do not change the way of queries, sparkProperties, sparkSqlProperties. Just change the values inside strings
```
Comment (sonalsagarwal) [inline tests/benchmark/snappy/tpcds/PerfRun.conf]: Before running the perf, we create a PerfRun.conf from the PerfRun.conf.template, so this file can be deleted.

File: tests/benchmark/snappy/dynamic/spark/PerfRun.conf [hunk 5]
```diff
@@ -0,0 +1,74 @@
+#Do not change the way of queries, sparkProperties, sparkSqlProperties. Just change the values inside strings
```
Comment (sonalsagarwal) [inline tests/benchmark/snappy/dynamic/spark/PerfRun.conf]: Before running the perf test, we create a PerfRun.conf from the PerfRun.conf.template, so this file can be deleted or renamed to PerfRun.conf.template

File: tests/benchmark/snappy/dynamic/smartConnector/PerfRun.conf [hunk 6]
```diff
@@ -0,0 +1,99 @@
+#Do not change the way of queries, sparkProperties, sparkSqlProperties. Just change the values inside strings
```
Comment (sonalsagarwal) [inline tests/benchmark/snappy/dynamic/smartConnector/PerfRun.conf]: Before running the perf test, we create a PerfRun.conf from the PerfRun.conf.template, so this file can be deleted or renamed to PerfRun.conf.template

File: tests/benchmark/snappy/dynamic/PerfRun_Jdbc.conf [hunk 7]
```diff
@@ -0,0 +1,102 @@
+#Do not change the way of queries, sparkProperties, sparkSqlProperties. Just change the values inside strings
```
Comment (sonalsagarwal) [inline tests/benchmark/snappy/dynamic/PerfRun_Jdbc.conf]: Before running the perf test, we create a PerfRun_Jdbc.conf from the PerfRun_Jdbc.conf.template, so this file can be deleted or renamed to PerfRun_Jdbc.conf.template

File: tests/benchmark/snappy/dynamic/PerfRun.conf [hunk 8]
```diff
@@ -0,0 +1,102 @@
+#Do not change the way of queries, sparkProperties, sparkSqlProperties. Just change the values inside strings
```
Comment (sonalsagarwal) [inline tests/benchmark/snappy/dynamic/PerfRun.conf]: Before running the perf test, we create a PerfRun.conf from the PerfRun.conf.template, so this file can be deleted or renamed to PerfRun.conf.template

File: cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/TableCreationJob.scala [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+import java.io.{File, FileOutputStream, PrintStream}
+
+import com.typesafe.config.Config
+import io.snappydata.benchmark.{TPCHColumnPartitionedTable, TPCHReplicatedTable}
+
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{SnappyJobValid, SnappyJobValidation, SnappySQLJob, SnappySession}
+
+/**
```
Comment (sonalsagarwal) [inline cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/TableCreationJob.scala]: Author details can be deleted.

File: cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/SparkApp.scala [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+package io.snappydata.benchmark.snappy.tpcds
+
+import java.io.{File, FileOutputStream, PrintStream}
+
+import io.snappydata.benchmark.snappy.tpch.QueryExecutor
+
+import org.apache.spark.sql.catalyst.util.fileToString
+import org.apache.spark.sql.{Row, SparkSession}
+
+/**
```
Comment (sonalsagarwal) [inline cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/SparkApp.scala]: Author details can be deleted

File: cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/SparkApp.scala [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+    //TPCDSQuerySnappyBenchmark.snappy = snSession
+    //TPCDSQuerySnappyBenchmark.execute(dataLocation, queries, true, queryPath)
+  }
+
+
+  def createPartitionedTables(sc: SparkSession, dataLocation: String,
+                                    partitionBy: String , tableName: String): Unit = {
+    val df = sc.sqlContext.read.parquet(s"$dataLocation/$tableName")
+    df.createOrReplaceTempView(tableName)
+    df.repartition(223, df(partitionBy)).createOrReplaceTempView(tableName)
```
Comment (swatisawant) [inline cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/SparkApp.scala]: Can you please explain why 223 value is used here?

File: cluster/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQuerySnappyBenchmark.scala [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+  var snappyRS: FileOutputStream = new FileOutputStream(
+    new File(s"Snappy_Results.out"))
+  var sparkRS : FileOutputStream = new FileOutputStream(
+    new File(s"Spark_Results.out"))
+
+  var snappyPS: PrintStream = new PrintStream(snappyRS)
+  var sparkPS: PrintStream = new PrintStream(sparkRS)
+
+  def setupTables(dataLocation: String, isSnappy: Boolean): Map[String, Long] = {
+    val props = Map("BUCKETS" -> "7")
```
Comment (swatisawant) [inline cluster/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQuerySnappyBenchmark.scala]: Why BUCKETS value is 7 here? is it hardcoded and we need to parameterize the same?

Comment (KishorBachhav) [reply cluster/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQuerySnappyBenchmark.scala]: Suyog, Can you please fix this.

Comment (svbhokare) [reply cluster/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQuerySnappyBenchmark.scala]: This is used by unit tests only & sufficient for test data used. If someone wants to change it, can directly replace with other number before running the test.

**Pull request merged**